### PR TITLE
rate limit nginx

### PR DESCRIPTION
**Pull request by emord**
Tested this on staging with ethan also hitting the server.

This will rate limit based on the server name so "cas.commcarehq.org" (which I think is what all phones will be hitting) It doesn't seem necessary to block the main site as that's only used via the web correct? if that's necessary, It'll also take some refactoring as we use the same base yml file for prod, softalyer and icds

@snopoke 

@gcapalbo @benrudolph

### CODE REVIEW

File: ansible/roles/nginx/vars/cas_ssl.yml [hunk 1]
```diff
@@ -11,6 +11,11 @@ nginx_sites:
     - name: "{{ deploy_env }}_cas_formplayer"
       hosts: formplayer
       port: "{{ formplayer_port }}"
+   limit_req_zones:
+    - name: "server_name"
+      zone: "$server_name"
+      size: "10m"
+      rate: "50r/s"
```
Comment (emord) [inline ansible/roles/nginx/vars/cas_ssl.yml]: This rate and burst are arbitrary and should likely be changed, but wasn't sure what we were expecting per second

**Comment by benrudolph** [comment]
nice this is cool. looks good to me once we figure out what we want as our rate

**Comment by benrudolph** [comment]
current rates on datadog peak around 90r/s. not sure how useful that is, but at leasts grounds some sort of estimate

**Comment by emord** [comment]
Forgot to mention that we could also limit this to certain endpoints by copying the nginx location configuration, but with a different limit_req and/or limit_req_zone. Could be useful if we don't want to limit phones updating the app or if we want restores and submissions to be limited differently

**Comment by benrudolph** [comment]
was looking at the nginx dashboard, but i forgot to filter out softlayer requests. once i limited to hqproxy0 i see 15r/s at the peak

**Comment by emord** [comment]
Realized we'd probably want to have one limit for the server and another per user/ip address so i added one of those as well

**Comment by snopoke** [comment]
Nice, was also thinking the same.

**Comment by emord** [comment]
@snopoke do you have any recommended limits for this? may be good to get this out before the rest of the phones are deployed

**Comment by snopoke** [comment]
Thanks for pinging - been meaning to come back to this. This is what I think we want:
* global - 200r/s (burst 0, nodelay = true)
* per IP - 5r/s (burst 0, nodelay = True) - this won't work for web requests but since this is only for request from the mobile I think it's OK.

**Comment by emord** [comment]
@snopoke thanks i updated the limits

**Comment by emord** [comment]
> this won't work for web requests but since this is only for request from the mobile I think it's OK.

do we want to limit for web? I was thinking that that would be unnecessary since there should be much less activity on web that mobile

**Comment by snopoke** [comment]
Yes I agree, no need to limit web requests. I was just noting that the per
IP rate limit would be to low for web requests.

Turns out it's quite useful to have mobile submitting to a separate site.

**Pull request merged**